### ISSUE TITLE
Unexpected type argument

### ISSUE THREAD

**Issue by vojtechkral** [kardeiz/serde_tuple#3]
Hi,
  I am getting the following error when trying to use `serde_tuple`:

    error[E0107]: wrong number of type arguments: expected 1, found 2
    
    16 | #[derive(Debug, Serialize_tuple, Deserialize_tuple)]
       |                                  ^^^^^^^^^^^^^^^^^ unexpected type argument

The code that causes this is copy-pasted from the doc:

```rust
#[derive(Serialize_tuple, Deserialize_tuple)]
pub struct Foo<'a> {
    bar: &'a str,
    baz: i32
}
```

but it also happens if the struct is empty.

**Comment by kardeiz** [comment]
Thanks for reporting!

I just tested and I'm not able to replicate this with the latest `rustc`s (stable and nightly) and serde/serde_derive `1.0.92`.

Could you provide some more information about your environment (versions for rustc, serde, serde_tuple, etc.) and/or the output of [cargo-expand](https://github.com/dtolnay/cargo-expand) for the failing struct?

**Comment by dtolnay** [comment]
It probably has to do with standard library types in the generated code, like Result. Macros are supposed to refer to fully qualified paths only, as the name may mean something different in the caller's code.

```rust
struct Result; struct Ok; struct Err;
struct Option; struct Some; struct None;

#[derive(Deserialize_tuple)]
...
```

**Comment by kotofond** [comment]
@kardeiz  @dtolnay is right, I'm facing the same issue when I have custom-defined Result in the same scope. If there is an easy fix available and you could fix it, it would be really great. Thanks

**Comment by vojtechkral** [comment]
@kotofond I was in fact using a custom `Result` type (or rather its `E` type). Thank you for commenting, I sort of forgot about this bugreport, sorry!

**Comment by kardeiz** [comment]
@vojtechkral @kotofond Thanks, and sorry about that! I had forgotten to fully qualify `Result` and `Result::Ok` for the `Deserialize` side of things. I have fixed this and released [`0.3.0`](https://crates.io/crates/serde_tuple/0.3.0).
